pallets/parachain-staking/src/tests.rs

```diff
@@ -1079,6 +1077,241 @@ fn payouts_follow_nomination_changes() {
 		});
 }
 
+#[test]
+/// Verify that only the top T::MaxNominatorsPerCollator are in top
+/// and the bottom nominators set is unbounded
+/// - MaxNominatorsPerCollator = 4
+/// - MaxCollatorsPerNominator = 4
+fn collator_nominator_sets_behave() {
```

Comment:
I hate to say it, but this should be split into a bunch of different isolated tests.

Reply:
I was organizing the tests like

```
runtime_method_works() {
   first_error_paths();
   second_error_path();
  ..
}
```

but I agree I should make a test for every path instead. This will improve readability (which I recognize is rough as is).

Reply:
I split it into four tests:
* `bottom_nominations_are_empty_when_top_nominations_not_full`
* `candidate_pool_updates_when_total_counted_changes`
* `only_top_collators_are_counted`
* `nomination_events_convey_correct_position`

Thanks for the suggestion, it is better organization! I'll follow up with the same changes to the other tests as well (not in this PR though).